### ISSUE TITLE
Increasing memory usage on CPU

### ISSUE THREAD

**Issue by varun-intel** [pytorch/pytorch#30388]
## 🐛 Bug

I am trying to run a small neural network on the CPU and am finding that the memory used by my script increases without limit. Since my script does not do much besides call the network, the problem appears to be a memory leak within pytorch.

The problem does not occur if I run the model on the gpu.

## To Reproduce

I am attaching a Dockerfile that builds my environment and a script (test.py) that reproduces the issue.
[https://gist.github.com/varun-intel/e36716c839c9ad5ced4dc08cd274d8e1](url)
[https://gist.github.com/varun-intel/08fe43f05533ad2613d97f19c9473d01](url)

The output of the script is:
https://gist.github.com/varun-intel/22456b07e7229215570d54875c841fe5
The memory usage continues to increase until the script crashes with an out of memory error.

## Expected behavior

The memory should increase until it reaches a reasonable steady state.

## Environment

Collecting environment information...
PyTorch version: 1.3.1
Is debug build: No
CUDA used to build PyTorch: 10.1.243

OS: Ubuntu 18.04.3 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.10.2

Python version: 3.5
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: TITAN Xp
Nvidia driver version: 430.26
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.3

Versions of relevant libraries:
[pip3] numpy==1.13.3
[conda] Could not collect

**Comment by varun-intel** [comment]
Has anyone had the chance to look into this issue? I want to point out that repeatedly running inference on a small model is an importance use-case in reinforcement learning models such as A3C.

**Comment by alanwang93** [comment]
same problem

**Comment by varun-intel** [comment]
I found that this only happened when I was using a virtualenv in my docker container.

**Comment by ngimel** [comment]
Can you provide a runnable example demonstrating the problem?

**Comment by varun-intel** [comment]
It's in my original post: please take a look at the dockerfile and the test.py script.

**Comment by ngimel** [comment]
Can it be related to #32008? Can you check if #32875 fixes it (unfortunately it is not merged yet)

**Comment by xsacha** [comment]
@varun-intel Are you still encountering this issue? Trying to track down increasing memory usage on another user machine that I can't replicate and feel like it may be related to running it via docker in some way.

**Comment by varun-intel** [comment]
You're right: the problem only showed up when I was using virtualenv inside docker. I just stopped doing that.
